cmake/onnxruntime_providers.cmake

```diff
@@ -1168,3 +1172,46 @@ if (onnxruntime_USE_ROCM)
           RUNTIME  DESTINATION ${CMAKE_INSTALL_BINDIR})
 
 endif()
+
+if (onnxruntime_USE_STVM)
+  add_definitions(-DUSE_STVM=1)
+
+  file (GLOB_RECURSE onnxruntime_providers_stvm_cc_srcs CONFIGURE_DEPENDS
+    "${ONNXRUNTIME_ROOT}/core/providers/stvm/*.h"
+    "${ONNXRUNTIME_ROOT}/core/providers/stvm/*.cc"
+    )
+  source_group(TREE ${ONNXRUNTIME_ROOT}/core FILES ${onnxruntime_providers_stvm_cc_srcs})
+  onnxruntime_add_static_library(onnxruntime_providers_stvm ${onnxruntime_providers_stvm_cc_srcs})
+
+  if ( CMAKE_COMPILER_IS_GNUCC )
+    target_compile_options(onnxruntime_providers_stvm PRIVATE -Wno-unused-parameter -Wno-missing-field-initializers)
+  endif()
+
+  target_include_directories(onnxruntime_providers_stvm PRIVATE
+          ${onnxruntime_STVM_HOME}/include
+          ${onnxruntime_STVM_HOME}/3rdparty/dlpack/include
+          ${onnxruntime_STVM_HOME}/3rdparty/dmlc-core/include
+          ${PYTHON_INLCUDE_DIRS})
+  onnxruntime_add_include_to_target(onnxruntime_providers_stvm onnxruntime_common onnx)
+
+  add_dependencies(onnxruntime_providers_stvm ${onnxruntime_EXTERNAL_DEPENDENCIES})
+
+  target_link_libraries(onnxruntime_providers_stvm PRIVATE
+      onnx
+      onnxruntime_common
+      onnxruntime_framework
+  )
+  if(${CMAKE_SYSTEM_NAME} MATCHES "Darwin")
+    target_link_libraries(onnxruntime_providers_stvm PRIVATE ${onnxruntime_STVM_HOME}/build/libtvm.dylib)
+  else()
```

Comment:
this assumes Unix/Linux systems only?
Does this work on Windows?

Reply:
Just now the target platform is Unix/Linux system and all tests were done on it only. In fact TVM can be supported on iOS or Windows. Now it is simply a draft for future development we did not check work on Windows.

Reply:
We've made a note in the documentation that this "preview" EP has only been tested in Linux. It is part of our plans to ensure this can run in Windows, but that capability won't be included in this PR.